sql/timetype_test.go

```diff
@@ -119,12 +119,12 @@ func TestTimeConvert(t *testing.T) {
 		{"40.134608", "00:00:40.134608", false},
 		{"401122.134612", "40:11:22.134612", false},
 		{"401122.134612585", "40:11:22.134613", false},
-		{"595959.99999951", "60:00:00", false},
-		{"585859.999999514", "58:59:00", false},
+		{"595959.99999951", "59:59:59.999999", false},
```

Comment:
@Hydrocharged can you rule on this? I know you were very particular about this, but I'm not clear what conversion semantics this is actually exercising, e.g.:

```
mysql> create table dates(a timestamp);
Query OK, 0 rows affected (0.02 sec)
mysql> insert into dates values ("595959.99999951");
ERROR 1292 (22007): Incorrect datetime value: '595959.99999951' for column 'a' at row 1
```

Reply:
@zachmu
You're using a `TIMESTAMP` rather than a `TIME`. Anyway, I should have spent the additional time on this back in the day, as there are some inconsistencies in the implementation. `TIME` by itself is equivalent to `TIME(0)` in MySQL. In that case, the original tests (this and the other unmodified ones below) are correct. However, for `TIME(6)`, the updated tests (this and the ones below) are correct. During testing, I must have accidentally used `TIME(0)` rather than `TIME(6)` for some of these tests.

Currently we don't support any parameters on the `TIME` type, making `TIME(0)` the expected type. In order to support microseconds, I instead made the implementation behavior that of `TIME(6)`, as it's the most precise of the `TIME` types. I can't quite remember why I didn't implement the parameter (probably one of those "_add it when someone needs it_" moments), but `TIME(6)` is how our `TIME` type behaves. So technically we should be returning `TIME(6)`, but it used to fail parsing. That is no longer the case anymore (not sure what changed).

@MathiasGr 
So we should change the `String()` function to return `"TIME(6)"` to make it explicit which type we're using. I think it's fine for now that `TIME` defaults to `TIME(6)` for us rather than `TIME(0)`, but add a `TODO` near the top for us to implement parameters on the `TIME` type. Additionally, check in `sql/type.go::ColumnTypeToType` if we're receiving the parameter, and if so, restrict the allowed types by throwing a **not yet supported** error for `TIME(0)` through `TIME(5)`, and one of the **invalid argument** errors any value >= 7.